rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/is_hvac_sys_fluid_loop_purchased_CHW.py

```diff
@@ -0,0 +1,47 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.utils.jsonpath_utils import find_exactly_one_with_field_value
+
+EXTERNAL_FLUID_SOURCE = schema_enums["ExternalFluidSourceOptions"]
+
+
+def is_hvac_sys_fluid_loop_purchased_chw(rmi_b, hvac_b_id):
+    """Returns TRUE if the fluid loop associated with the cooling system associated with the HVAC system is attached to an external purchased chilled water loop. Returns FALSE if this is not the case.
+    Parameters
+    ----------
+    rmi_b : json
+        RMD at RuleSetModelInstance level
+    hvac_b_id : str
+        The HVAC system ID.
+    Returns
+    -------
+    bool
+        True: the fluid loop associated with the cooling system associated with the HVAC system is attached to an external purchased cooling loop
+        False: otherwise
+    """
+    is_hvac_sys_fluid_loop_purchased_chw_flag = False
+    purchased_cooling_loop_list_b = []
+
+    external_fluid_sources = rmi_b.get("external_fluid_source")
+    if external_fluid_sources:
+        for external_fluid_source in external_fluid_sources:
+            if (
```

Comment:
Should we be raising an exception if the _external_fluid_source_  does not have a _loop_?  If that is the case, this code could be:
```
if external_fluid_source.get("type") == EXTERNAL_FLUID_SOURCE.CHILLED_WATER:
    loop = getattr_(external_fluid_source, "ExternalFluidSource", "loop")
    purchased_cooling_loop_list_b.append(loop)
```